rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,110 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie as incidents are intended to be used for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured in Teleport's config yaml file. The required fields will be added to a 'plugins' section containing the required information to interact with both Teleport access and the Opsgenie API.
+
+```
+plugins:
+    opsgenie:
```

Comment:
When the plugin runs standalone, how does it connect to the cluster? It should have auth address and identity file in the settings, right? If we're using teleport.yaml fileconfig format, there's already auth_server field but we should be able to pass it identity somehow.

Reply:
Also, if we're running inside Teleport, can the plugin be just a separate service (like, app service, database service, etc.) and retrieve its identity same way as any other agent? cc @espadolini what do you think?

Reply:
I like the idea of having this to be a separate service :+1: 
We would need to create a join token and a new System Role for plugins
That would decrease the setup time because you no longer need to create a user and a role to interact with the API

Reply:
:nail_care:
Should we allow for multiple instances of the same plugin?
I don't think this would be required now but could allow us to have a generic HTTP plugin in the future that works with multiple endpoints.